### PR TITLE
feat(addon-editor): support hilite groups

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/addon-editor/components/toolbar-tools/group/group.component.ts [hunk 1]
```diff
@@ -1,19 +1,40 @@
...
 import {TuiLanguageEditor} from '@taiga-ui/i18n';
 import {Observable} from 'rxjs';
 import {distinctUntilChanged, map} from 'rxjs/operators';
 
 @Component({
     selector: `tui-editor-group-tool`,
     templateUrl: `./group.component.html`,
+    styleUrls: [`../tools-common.less`],
     changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.Emulated,
```
Comment (waterplea) [inline projects/addon-editor/components/toolbar-tools/group/group.component.ts:25]: What is it for?

File: projects/addon-editor/components/toolbar-tools/group/group.component.ts [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
 })
 export class TuiEditorGroupToolComponent {
+    @Input()
+    @tuiDefaultProp()
+    colors: ReadonlyMap<string, string> = this.defaultOptions.colors;
+
+    @Input()
+    hilite = false;
```
Comment (waterplea) [inline projects/addon-editor/components/toolbar-tools/group/group.component.ts:33]: Maybe just rely on `colors` being non-empty map?

**Review by waterplea — COMMENTED** [review]
Do we have that bucket icon in proprietary package? Also I think we should combine those into 1 page "Groups" with 3 examples:

Draggable groups
Hilite groups
Nested groups

File: projects/addon-editor/extensions/group/group.extension.ts [hunk 3]
```diff
@@ -73,6 +99,31 @@ export const createGroupExtension = (
 
                         return commands.insertContent(result);
                     },
+                setGroupHilite:
+                    (color: string) =>
+                    ({state, editor, dispatch}) => {
```
Comment (waterplea) [inline projects/addon-editor/extensions/group/group.extension.ts:104]: I'm not sure I understand this function 🤔 Could you clarify it for me?

**Comment by waterplea** [comment]
Probably makes sense to add `transparent` or `Remove color` option.

File: projects/addon-editor/components/toolbar-tools/group/group.component.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/components/toolbar-tools/group/group.component.ts]: fixed

File: projects/addon-editor/components/toolbar-tools/group/group.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-editor/components/toolbar-tools/group/group.component.ts]: fixed

File: projects/addon-editor/extensions/group/group.extension.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-editor/extensions/group/group.extension.ts]: update comments

**Review by waterplea — APPROVED** [review]

**Pull request merged**